environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/decorator/DecoratedWithPrivateConstructorTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.weld.environment.se.test.decorator;
+
+
+import javax.enterprise.inject.spi.DeploymentException;
+
+import org.jboss.weld.environment.se.Weld;
+import org.jboss.weld.environment.se.WeldContainer;
+import org.jboss.arquillian.container.se.api.ClassPath;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.BeanArchive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+public class DecoratedWithPrivateConstructorTest {
+
+    @Deployment
+    public static Archive<?> createTestArchive() {
+        return ClassPath.builder().add(ShrinkWrap.create(BeanArchive.class)
+                                               .addPackage(Bravo.class.getPackage()))
+                .build();
+    }
+
+    @Test
+    public void privateConstructorTest() {
+        try {
+            WeldContainer container = new Weld().initialize();
+            Assert.fail();
+        } catch (DeploymentException e) {
+            Assert.assertTrue(e.getMessage().contains("Bean class which has decorators should declare at least one non private constructor"));
```

Comment:
Assertion on an exact `String` in exception is a bad idea.
You can either simply `catch (DeploymentException e)` and do nothing (because you `fail()` it otherwise and the point here is not to get NPE) or you could use the `@Test(expected=DeploymentException.class)`.